### PR TITLE
Cspl 674 remove Spark from codebase

### PR DESCRIPTION
**Pull request by mgaldino-splunk**
This PR Removes Spark from the codebase.

Unit tests and Integration tests passing.

*The commits will be squashed once we're about to merge.

### CODE REVIEW

File: docs/CustomResources.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ Enterprise resources, including: `Standalone`, `LicenseMaster`,
 | clusterMasterRef  | [ObjectReference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#objectreference-v1-core) | Reference to a Splunk Operator managed `ClusterMaster` instance (via `name` and optionally `namespace`) to use for indexing |
 | serviceAccount | [ServiceAccount](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) | Represents the service account used by the pods deployed by the CRD |
 
+
```
Comment (akondur) [inline docs/CustomResources.md:122]: Do we need this extra line?

File: deploy/examples/spark/default.yaml [hunk 2]
```diff
@@ -1,5 +0,0 @@
-apiVersion: enterprise.splunk.com/v1
```
Comment (akondur) [inline deploy/examples/spark/default.yaml:1]: Does this remove the folder spark?

File: build/make_bundle.sh [hunk 3]
```diff
@@ -151,6 +138,4 @@ yq w $OLM_CATALOG/splunk/$VERSION/splunk.v${VERSION}.clusterserviceversion.yaml
 yq w $OLM_CATALOG/splunk/splunk.package.yaml packageName "splunk-certified" > $OLM_CERTIFIED/splunk/splunk.package.yaml
 
 # Mac OS expects sed -i '', Linux expects sed -i''. To workaround this, using .bak
-sed -i.bak "s,$DOCKER_IO_PATH/spark,$REDHAT_REGISTRY_PATH/spark,g" $OLM_CERTIFIED/splunk/splunk.v${VERSION}.clusterserviceversion.yaml
```
Comment (akondur) [inline build/make_bundle.sh:154]: Are we sure this has no other side effects on **$OLM_CERTIFIED/splunk/splunk.v${VERSION}.clusterserviceversion.yaml**?

File: pkg/splunk/common/util.go [hunk 4]
```diff
@@ -202,7 +202,7 @@ func CompareSortedStrings(a []string, b []string) bool {
 // GetIstioAnnotations returns a map of istio annotations for a pod template
 func GetIstioAnnotations(ports []corev1.ContainerPort) map[string]string {
 	// list of ports within the deployments that we want istio to leave alone
-	excludeOutboundPorts := []int32{8089, 8191, 9997, 7777, 9000, 17000, 17500, 19000}
+	excludeOutboundPorts := []int32{8089, 8191, 9997}
```
Comment (akondur) [inline pkg/splunk/common/util.go:205]: Are the ports which are removed here for SPARK only?

File: pkg/splunk/enterprise/names.go [hunk 5]
```diff
@@ -60,13 +60,13 @@ const (
...
 	// command for the init container with DFC
-	commandForDfc = "cp -r /opt/jdk /mnt && cp -r /opt/spark /mnt"
+	//commandForDfc = "cp -r /opt/jdk /mnt"
 
 	// command for init container on a standalone without DFC
 	commandForStandaloneSmartstore = "mkdir -p /opt/splk/etc/apps/splunk-operator/local && ln -sfn  /mnt/splunk-operator/local/indexes.conf /opt/splk/etc/apps/splunk-operator/local/indexes.conf && ln -sfn  /mnt/splunk-operator/local/server.conf /opt/splk/etc/apps/splunk-operator/local/server.conf"
 
 	// command for init container with Smartstore enabled and DFC enabled
-	commandForDfcAndSmartstore = commandForDfc + commandMerger + commandForStandaloneSmartstore
+	//commandForDfcAndSmartstore = commandForDfc + commandMerger + commandForStandaloneSmartstore
```
Comment (akondur) [inline pkg/splunk/enterprise/names.go:69]: Are we commenting here vs removing?

File: deploy/examples/spark/default.yaml (hunk 2 above)
Comment (mgaldino-splunk) [reply deploy/examples/spark/default.yaml:1]: Yes, we're removing the entire folder.

File: docs/CustomResources.md (hunk 1 above)
Comment (mgaldino-splunk) [reply docs/CustomResources.md:122]: Thanks. Removing it in the next commit.

File: pkg/splunk/common/util.go (hunk 4 above)
Comment (mgaldino-splunk) [reply pkg/splunk/common/util.go:205]: Yes, we're removing 7777, 9000, 17000, 17500, 19000 across the code.

File: pkg/splunk/enterprise/names.go (hunk 5 above)
Comment (mgaldino-splunk) [reply pkg/splunk/enterprise/names.go:69]: Good catch @akondur. We'll remove it on next commit.

File: build/make_bundle.sh (hunk 3 above)
Comment (mgaldino-splunk) [reply build/make_bundle.sh:154]: Good point. Do you think I should run any specific test?
Maybe updating a CRD version and running it again?

Comment (akondur) [reply build/make_bundle.sh:154]: Running a **"make generate"** and comparing should be good.

Comment (mgaldino-splunk) [reply build/make_bundle.sh:154]: Resolved.

File: pkg/splunk/enterprise/util.go [hunk 6]
```diff
@@ -224,11 +223,10 @@ func ApplySmartstoreConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 	return SplunkOperatorAppConfigMap, configMapDataChanged, nil
 }
 
-//  setupInitContainer modifies the podTemplateSpec object to incorporate support for DFS.
-func setupInitContainer(podTemplateSpec *corev1.PodTemplateSpec, sparkImage string, imagePullPolicy string, commandOnContainer string) {
-	// create an init container in the pod, which is just used to populate the jdk and spark mount directories
+//  setupInitContainer modifies the podTemplateSpec object
+func setupInitContainer(podTemplateSpec *corev1.PodTemplateSpec, Image string, imagePullPolicy string, commandOnContainer string) {
 	containerSpec := corev1.Container{
-		Image:           sparkImage,
```
Comment (smohan-splunk) [inline pkg/splunk/enterprise/util.go:231]: Had an offline chat with @mgaldino-splunk about this, understand the InitContainer is used by SmartStore code. cc: @sgontla , @pdhanoyasplunk

**Review by smohan-splunk — APPROVED** [review]

**Review by akondur — APPROVED** [review]

**Comment by sgontla** [comment]
As I am understanding, a couple of the init container properties should be simple and they should run to completion. Was chatting with Marcus, and with the tests he did, he doesn't notice any difference with initialization time, whether it is Spark Or Splunk. So, we are good. 
Having said that, I think we need to explore a bit more and understand how exactly things are working in the case of the init container.

**Review by sgontla — APPROVED** [review]

File: pkg/splunk/common/util.go [hunk 7]
```diff
@@ -202,7 +202,7 @@ func CompareSortedStrings(a []string, b []string) bool {
 // GetIstioAnnotations returns a map of istio annotations for a pod template
 func GetIstioAnnotations(ports []corev1.ContainerPort) map[string]string {
 	// list of ports within the deployments that we want istio to leave alone
-	excludeOutboundPorts := []int32{8089, 8191, 9997, 7777, 9000, 17000, 17500, 19000}
```
Comment (gaurav-splunk) [inline pkg/splunk/common/util.go:205]: so these ports(7777, 9000, 17000, 17500, 19000) were only used by Spark?

**Review by gaurav-splunk — APPROVED** [review]

File: pkg/splunk/common/util.go (hunk 7 above)
Comment (mgaldino-splunk) [reply pkg/splunk/common/util.go:205]: Yes, that's correct.

**Pull request merged**